rust/connlib/snownet/src/node.rs

```diff
@@ -143,6 +146,10 @@ where
         for allocation in self.allocations.values_mut() {
             allocation.refresh(now);
         }
+
+        for connection in self.connections.established.values_mut() {
+            connection.created_at = now;
+        }
```

Comment:
This is kind of critical. Upon re-connect, we need to reset this timestamp to ensure we have a new grace-period and don't immediately emit all (new) relay candidates. Once https://github.com/firezone/firezone/pull/4585 is merged, I'll try to add a unit-test for this.